tests/ui/consts/miri_unleashed/mutable_references_err.rs

```diff
@@ -56,10 +63,15 @@ const POINTS_TO_MUTABLE2: &i32 = unsafe { &*MUTABLE_REF };
 
 const POINTS_TO_MUTABLE_INNER: *const i32 = &mut 42 as *mut _ as *const _;
 //~^ ERROR: mutable pointer in final value
+//~| WARNING this was previously accepted by the compiler
+
 const POINTS_TO_MUTABLE_INNER2: *const i32 = &mut 42 as *const _;
 //~^ ERROR: mutable pointer in final value
+//~| WARNING this was previously accepted by the compiler
+
 const INTERIOR_MUTABLE_BEHIND_RAW: *mut i32 = &UnsafeCell::new(42) as *const _ as *mut _;
 //~^ ERROR: mutable pointer in final value
+//~| WARNING this was previously accepted by the compiler
```

Comment:
It looks like we have some places where we were already generating an error but now we report the future incompatibility warning. That's perhaps a bit confusing but I don't think it needs to block the PR.

Reply:
Note that this file uses `-Zunleash-the-miri-inside-of-you`. I am not sure if this can even be triggered on stable.

Reply:
Could just not emit the future incompat lint if unleash mode is on

Reply:
We want the error in unleash mode, it's the only way unleash mode can be sound.

Reply:
> Could just not emit the future incompat lint if unleash mode is on

As a variation on the above, I was musing that if *any* future-incompat lint is set to deny or forbid, then we could stop emitting that boilerplate about the thing being potentially rejected in a future version of the compiler, just to cut down on the diagnostic flooding...